app/controllers/apply_job_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ApplyJobController < ApplicationController
+
+  def new
+    url = request.url 
+    job_id = url.scan(/[0-9]{1,8}$/).last.to_i    
+    if Job.where(id: job_id).blank?
+      render 'errors/file_not_found'
```

Comment:
Có thể sử dụng guard clause
```suggestion
    return render 'errors/file_not_found' unless Job.exist?(id: job_id).blank?
```